### ISSUE TITLE
Speed and offset settings changing on their own

### ISSUE THREAD

**Issue by Plisko1** [Courseplay/courseplay#2806]
I'm not sure if this is a bug or me screwing up but I am having an issue where the settings for the vehicle are not staying when I change them.  

Issue 1:  When tipping a chaff trailer at the bunkers, the course seems to use the reverse speed setting (3 mph) to determine the speed of dumping the load forward through the bunker silo at a tip trigger... I can change the reverse speed to 6 mph as it starts to dump and it will go faster, letting me spread the whole length of the silo.  However, the minute it arrives with a new load an gets to the tip trigger again, it resets to 3 mph and I need to change it before it starts dumping.  I can't seem to get it to stay at 6 mph.

Issue 2: Is similar but with the tool offset. I have a merged course with the pickup and drop off of bales.   I adjusted the offset with the round bail pickup trailer to what I wanted.  It stays when I turn on the driver.  Turn him off and it stays.  Turn him back on and it changes back to the old setting.  I can then basically toggle between the two settings by turning him on or off.  I can't get it to just stay there.  I thought maybe it had to do with how I saved the courses... but I have tried loading them back in and changing the setting and saving the merged courses as a new save with the setting I want... it doesn't stop the behavior.

**Comment by Plisko1** [comment]
I think this might be me.  I am now seeing evidence that everything set on the tractor when a course is first saved is the setting it keeps and if the settings are changed on the tractor it still reverts to the course settings when the course is started.  I am sorry if I missed this in the manual.  I love this mod but I am still working out the specifics.

**Comment by pops64** [comment]
No it is not you. Fwd speed through the BGA is set so it empties its entire load over 80% length of the BGA. Tool offest is automatically set if the tool already has a tool offest associated with it. Most of the vinalla and dlc has this if the tool needs it e.g. ploughs. Also all of these settinga are specific to the vehicle. And not shared

**Comment by Plisko1** [comment]
The gotcha for me with the offset problem is that if you record a drive around  a grass field in order to create a haying course, you can't accidentally have the tool offset set to anything or it will carry over to any course created with that field outline.  Then, when you set the tool later, it reverts back to that burned in setting every time the course is restarted.

**Comment by pops64** [comment]
It shouldn't just rember that the center of the course is where the tractor is at the time of recording. If you recored based off tool center then you need to add negative tool offset to any tool that does not share this offset. Also have you checked out the use course has field boundary in the course gen this way you only have to recored the edge of the field and CP will do all the other course through course generation

**Comment by Plisko1** [comment]
That's what I was doing.  I am merging several fields for wrapping grass bales...  I did it by driving the boundary around all of them and then generating the course based on that boundary.   It works great. I suspected that when I recorded the boundary, I had the offset on by accident so, when I generated the course from that boundary the offset carried over.   So, no matter what the tractors settings are for the tool, when you start the driver it reverts to that other number.  As I turn the driver on or off it toggles between the number I set... and this mysterious other number, which seems to usually be 2.4 M to the left.  At first it was happening with both the bale wrapper and the bale collector.  The second time when I was more careful it was not happening with the wrapper but still happening with the collector.  I would set the tractor to 5m tool offset but when I started the course, it would always revert back to 2.4m.  It accepts changes again after it is driving but it is a bit of a pain.

I am also still not understanding why the trucks dump at 3mph no matter what I set the reverse speed to.  They never make it to the end of the bunker at mph.  

If things seem unusual and you need logs let me know. I am on the goldcrest valley map with Courseplay version 5.3.0 using the Mac OS through Steam.

**Comment by pops64** [comment]
I'd would recommend upgraded to the dev version, we are usually very careful with pushing a stable version but every1 once in a while a typo creeps in so the warning about using it still does apply but there are bug fixes and other such stuff that doesn't get pushed to giants has often. For the collector and wrapper if you are using vanilla then it has a tool offset associated with it, this should be spot on for what you want.(Bales are kinda a tricky has they are randomly dropped and our system currently is static for these two tools) Has for the BGA if your reversing into it then it is a completely different setup, this stops 10 or 20%(Don't Remember) into the length of the bunker and dumps the entire load there, it might move slowly forward if the heap is full. Also check out this https://courseplay.github.io/courseplay/en/manual/index.html#hud_drivingSettings It is still accurate minus the stuff about lane offset unless you do it the manual way. Also having offset can't effect your field boundary(It is not recored in any such that gets saved to the course just the vehicles.xml found in your savegame folder) . Just make sure that the course is always recored from the center of the vehicle. Even if the part of the vehicle your driving center line is based off is not

**Comment by Plisko1** [comment]
Thanks... for the advice.  I will consider the dev version.  I am playing with some mods but most of them are vehicles and small tweaks.    The problem with the wrapper/bail collector in my case is that the wrapper needs to be offset 2.5m left to pick up after the bailer drops it on the center line.  Then it drops the wrapped bale a little offset from center in such a way that the bale collector trailer following then needs even more offset, usually closer to 5m to compensate for it's own offset plus the offset bale.    I don't see any associated offset that makes this work with both tools traveling together... but I will look closer.

As for the BGA, sorry if I was confusing....  I am not reversing in.  I am driving straight through the bunker.  When the truck with a driver hits the tip trigger it begins slowly moving forward as it unloads at a fixed speed.  This default speed doesn't make it to the end.  It finishes unloading about 3/4 of the way which creates an uneven situation as it builds up.  After some experimenting I noticed that the speed the truck moves during the forward traveling tip changes if I change the "reverse speed" setting.  So my ideal speed for dumping forward is 6mph not 3mph... for example.  It will go at 6mph only if I change it manually right after the driver hits the tip trigger as it enters the silo.   The problem is that It always resets back to 3mph when it tips again on the next trip... and I set it manually again... and so on.  It's kind of a hands on situation to make it work the way I need.

I appreciate your time.  I am still a little new at this and I'm  insecure about what is operator error and what might be a real issue.
